src/msg.c

```diff
@@ -1065,7 +754,1282 @@ void draw_message_window(MessagePrintState* printer) {
+    s16 spAE;
+    u16 spB6;
+    u8 spB8;
+    f32 spBC;
+    s32* spC0;
+    MessageCharset* msgCharset;
+    f32 temp_f10;
+    f32 temp_f20;
+    f32 temp_f20_3;
+    f32 temp_f22_2;
+    f32 temp_f24;
+    f32 bubbleX;
+    f32 bubbleY;
+    f32 temp_f28;
+    f32 temp_f30;
+    f32 temp_f8;
+    f32 temp_f8_5;
+    f64 temp_f0;
+    f64 temp_f0_2;
+    f64 temp_f2_2;
+    f64 temp_f20_2;
+    f64 temp_f22;
+    f64 temp_f2;
+    f64 temp_f2_3;
+    f64 temp_f2_6;
+    f64 temp_f2_7;
+    f64 temp_f4;
+    f64 temp_f4_4;
+    f64 temp_f4_5;
+    f64 temp_f6_2;
+    s32 temp_s0_7;
+    s32 temp_s0_8;
+    s32 temp_a2;
+    s32 temp_s1_5;
+    s32 i;
+    s16 temp_s3;
+    s16 temp_s4;
+    s32 windowSizeX;
+    s32 windowSizeY;
+    s16 temp_s5;
+    s16 temp_s6;
+    u32 temp_a0_49;
+    s32 animIdx;
+    u8 temp_v0_27;
+    u8 temp_v0_28;
+    UnkImage14* msgVarImage;
+    UnkImage14* temp_t4;
+    s32 phi_a0;
+    s32 straightWidth;
+    s32 curveWidth;
+    s32 height;
+    s32 frameX;
+    s32 frameY;
+    s32 frameSizeX;
+    s32 frameSizeY;
+    s32 frameFading;
+    s32 frameAlpha;
+    u16 fading;
+    s32 phi_s0_5;
+    s32* phi_s6;
+    s8 phi_s2_4;
+    s32 phi_s3_2;
+    s32 phi_v0_3;
+    s32 phi_a0_4;
+    s16 phi_t5;
+    s16 phi_s0_6;
+    s16 phi_t3;
+    s32 dbPos;
+    s16 charPosX;
+    s16 charPosY;
+    s16 phi_s2_5;
+    s32 phi_s0_7;
+    s32 phi_s1_8;
+    s8 varImgHasBorder;
+    s16 t1;
+    s16 t2;
+    f32 z1;
+    f32 z2;
+    f32 temp1;
+    f32 temp2;
+
+    msg_drawState = &D_80155D20;
+    msg_drawState->printBuffer = printer->printBuffer;
+    msg_drawState->printModeFlags = 0;
+    if (gMasterGfxPos != D_80151338) {
+        msg_reset_gfx_state();
+    }
+    msg_drawState->printModeFlags |= 0x100 | 1;
+    sp8E = 0xFF;
+    if (flag & 1) {
+        sp8E = alpha;
+    }
+    msg_drawState->textStartPos[0] = 0;
+    msg_drawState->textStartPos[1] = 0;
+    msg_drawState->clipX[0] = 0;
+    msg_drawState->clipY[0] = 0;
+    msg_drawState->clipX[1] = SCREEN_WIDTH - 1;
+    msg_drawState->clipY[1] = SCREEN_HEIGHT - 1;
+    msg_drawState->msgScale.x = 1.0f;
+    msg_drawState->msgScale.y = 1.0f;
+    msg_drawState->unk_29 = 0;
+    msg_drawState->framePalette = 0;
+    msg_drawState->unk_2C = 0;
+    msg_drawState->visiblePrintedCount = 0;
+    msg_drawState->centerPos = 0;
+    msg_drawState->fontVariant = 0;
+    msg_drawState->currentPosX = 0;
+    msg_drawState->nextPos[0] = 0;
+    msg_drawState->nextPos[1] = 0;
+    msg_drawState->font = 0;
+    msg_drawState->effectFlags = 0;
+    msg_drawState->nextCounter = 0;
+    msg_drawState->textColor = 0;
+    sp96 = 0xFF;
+    spAE = 0xFF;
+    spB6 = 0;
+    spB8 = 0;
+    sp9E = -1;
+
+    if (flag & 2) {
+        msg_drawState->font = 1;
+    }
+
+    if (flag & 4) {
+        gDPSetScissor(gMasterGfxPos++, G_SC_NON_INTERLACE, 0, 0, SCREEN_WIDTH - 1, SCREEN_HEIGHT - 1);
+    }
+
+    sp80bool = FALSE;
+    msg_drawState->drawBufferPos = 0;
+
+    do {
```

Comment:
sp80bool is set to 0 right before this loop. Will it not match as `while (!sp80bool)` instead of a do while?

Reply:
It does. I didn't have this open at first to test this myself.